### PR TITLE
WIP: df manipulations examples

### PR DESCRIPTION
**Pull request by r-brink**
Initial commit to add examples and docs around Dataframe manipulations

### CODE REVIEW

File: user_guide/src/examples/df_manipulations/casting.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from .dataset import df
+
+df["a"] = df["a"].cast(float)
```
Comment (ritchie46) [inline user_guide/src/examples/df_manipulations/casting.py:3]: Maybe mention that we can also cast with polars dtypes like `pl.Float32` or `pl.Float64`

File: user_guide/src/howcani/df/common-manipulations.md [hunk 2]
```diff
@@ -1 +1,122 @@
-# Filter
+# Common manipulations
+
+Like other DataFrame libaries, Polars provides a wide range of common functions to
+manipulate a Datframe. Users that are familiar with Dataframes will see many
```
Comment (carnarez) [inline user_guide/src/howcani/df/common-manipulations.md:4]: DataFrame

File: user_guide/src/howcani/df/common-manipulations.md [hunk 3]
```diff
@@ -1 +1,122 @@
...
+similarities.
```
Comment (carnarez) [inline user_guide/src/howcani/df/common-manipulations.md:5]: ... "with `Pandas` or `R` implementations."

File: user_guide/src/howcani/df/common-manipulations.md [hunk 4]
```diff
@@ -1 +1,122 @@
...
+```
+
+```text
+{{#include ../../outputs/df_manipulations/null_count.txt}}
+```
+
+## Sort columns
+
+```python
+df.sort('a', reverse=True)
```
Comment (carnarez) [inline user_guide/src/howcani/df/common-manipulations.md:97]: If we try to keep things formatted via `black`, this example should be `df.sort("a", reverse=True)` (note the `"`).

File: user_guide/src/examples/df_manipulations/casting.py (hunk 1 above)
Comment (r-brink) [reply user_guide/src/examples/df_manipulations/casting.py:3]: added as a TODO

**Pull request merged**